pkg/kubecost/summaryallocation_json.go

```diff
@@ -0,0 +1,116 @@
+package kubecost
+
+import (
+	"math"
+	"time"
+)
+
+// SummaryAllocationResponse is a sanitized version of SummaryAllocation, which
+// formats fields and protects against issues like mashaling NaNs.
+type SummaryAllocationResponse struct {
+	Name                   string    `json:"name"`
+	Start                  time.Time `json:"start"`
+	End                    time.Time `json:"end"`
```

Comment:
`type AllocationJSON` types these as `string` rather than `time.Time`. I'm not sure why this is, but maybe there was a good reason for it and we should follow the convention here?

Reply:
I will reach out to @saweber to ask about that. But `AllocationJSON` also uses the `Window` type, which encodes `time.Time`s, so even internal to `AllocationJSON` there are multiple conventions going on.

Trying to let the standard JSON marshaler do its thing, unless strictly necessary to protect against errors.

Reply:
`Window` actual has a custom marshaler which ensures the `time.RFC3339` format as well as our weird string `"null"` return: https://github.com/opencost/opencost/blob/0ada30f9653a6fdcfa112c4f142c96aa7bb1fd17/pkg/kubecost/window.go#L482-L496

Reply:
I did not do anything special with `AllocationJSON` other than rely on the default JSON encoding for `type Window`. I naively relied on Window marshaling itself correctly.

@michaelmdresser interesting find!

Reply:
Wowow very interesting. (Makes sense, of course... the Window fields are private. SMDH.)